apps/examples/connect_test/connect_test_client.c

```diff
@@ -0,0 +1,676 @@
+		ret = _tcp_recv_data(sockfd, size);
+		if (ret < 0) {
+			CT_LOGE(TAG, "receive fail size %d ret %d code %d", size, ret, errno);
+			close(sockfd);
+			return -1;
+		}
+		CT_LOG(TAG, "send ACK data size %d", WT_ACK_SIZE);
+		ret = _tcp_send_data(sockfd, WT_ACK_SIZE);
+		if (ret < 0) {
+			CT_LOGE(TAG, "send fail size %d ret %d code %d", WT_ACK_SIZE, ret, errno);
+			return -1;
+		}
+		CT_LOG(TAG, "sleep 500ms -> before closing connection, allows receiving of data");
+		usleep(500000);
+		close(sockfd);
+		CT_LOG(TAG, "tcp data send finish");
+	}
+	return 0;
+}
+
+static int broadcast_receive_init(int *sock) {
+	if ((*sock = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
+		CT_LOGE(TAG, "Socket creation failed, error description-value is : %s-%d", strerror(errno), errno);
+		return -1;
+	}
+
+	int broadcast = 1;
+	if (setsockopt(*sock, SOL_SOCKET, SO_BROADCAST, &broadcast, sizeof(broadcast)) < 0)
+	{
+		CT_LOGE(TAG, "Error in setting Broadcast option, error description-value is : %s-%d", strerror(errno), errno);
+		return -1;
+	}
+
+	/* Set send and receive timeout values */
+	struct timeval tv;
+	tv.tv_sec = 2;
+	tv.tv_usec = 0;
+	if (setsockopt(*sock, SOL_SOCKET, SO_RCVTIMEO, (struct timeval*)&tv, sizeof(struct timeval)) < 0) {
+		CT_LOGE(TAG, "Set time out failed for recvfrom");
+	}
+
+	struct sockaddr_in recv_addr;
+	memset(&recv_addr, 0, sizeof(recv_addr));
+
+	recv_addr.sin_family = AF_INET;
+	recv_addr.sin_port = htons(BROADCASTING_PORT);
+	recv_addr.sin_addr.s_addr = INADDR_ANY;
+
+	if (bind(*sock, (struct sockaddr*)&recv_addr, sizeof(recv_addr)) < 0)
+	{
+		CT_LOGE(TAG, "Error in BINDING , error description-value is : %s-%d", strerror(errno), errno);
+		return -1;
+	}
+	return 0;
+}
+
+static int broadcast_receive(int sock, char* msg, int msg_len)
+{
+	int nbytes = 0;
+	unsigned int len = sizeof(struct sockaddr_in);
+	int retry_count = 10;
+	struct sockaddr_in sender_addr;
+	while (retry_count--) {
+		nbytes = recvfrom(sock, msg, msg_len, 0, (struct sockaddr*)&sender_addr, &len);
+		if (nbytes > 0) {
+			CT_LOG(TAG, "received message: %s", msg);
+			char *broadcast_mac, *broadcast_ip;
+			broadcast_mac = strtok(msg, " ");
+			broadcast_ip = strtok(NULL, " ");
+			CT_LOG(TAG, "The stored mac of peer is %s", g_recv_mac_buf);
+			if(!strncmp(g_recv_mac_buf, broadcast_mac, strlen(g_recv_mac_buf))) {
+				strncpy(g_tcp_server_address, broadcast_ip, strlen(broadcast_ip)+1);
+				CT_LOG(TAG, "The mac of peer is %s and ip of peer is %s",broadcast_mac, broadcast_ip);
+				CT_LOG(TAG, "server ip %s", g_tcp_server_address);
+				break;
+			} else {
+				continue;
+			}
+		} else {
+			if (errno == EWOULDBLOCK) {
+				CT_LOGE(TAG, "timeout trying again, error %s-%d", strerror(errno), errno);
+				continue;
+			} else {
+				CT_LOGE(TAG, "Receving broadcast message failed");
+				return -1;
+			}
+		}
+	}
+	return 0;
+}
+
+int connect_test_run_client(void)
+{
+	int conn = 0;
+	/*  connect to softAP */
+	CT_LOG(TAG, "connect AP");
```

Comment:
how about to make new function for connecting wi-fi. it'll increase readability

Reply:
Changes done as suggested